syntax_and_semantics/literals/symbol.md

```diff
@@ -1,45 +1,62 @@
 # Symbol
 
-A [Symbol](http://crystal-lang.org/api/Symbol.html) is a constant that is identified by a name without you having to give it a numeric value.
+A [Symbol](http://crystal-lang.org/api/Symbol.html) represents a unique name inside the entire source code.
+It is internally implemented as a constant with a numeric value of type `Int32`.
+
+Symbols are interpreted at compile time and cannot be created dynamically. The only way to create a Symbol is by using a symbol literal, denoted by a colon (`:`) followed by an identifier. The identifier may be enclosed in double quotes (`"`).
 
 ```crystal
-:hello
-:good_bye
+:unquoted_symbol
+:"quoted symbol"
+```
+
+A double-quoted identifier can contain any unicode character including white spaces and accepts the same escape sequences as a [string literal](./string.html), yet no interpolation.
 
-# With spaces and symbols
-:"symbol with spaces"
+An unquoted identifier can contain alphanumeric characters, underscore (`_`) or characters with a code point greater than `159`(`0x9F`). It must not start with a number and may end with an exclamation mark (`!`) or question mark (`?`).
```

Comment:
I would mention that these are the same rule for naming methods. (either before or after the detailed explanation)